src/Cron_Event_Command.php

```diff
@@ -463,6 +426,56 @@ protected static function get_cron_events() {
 
 	}
 
+	/**
+	 * Fetches an array of scheduled cron events selected by the user.
+	 *
+	 * @param array $args       A list of event names
+	 * @param array $assoc_args An associative list of CLI parameters
+	 *
+	 * @return array|WP_Error An array of objects, or a WP_Error object is there are no events scheduled.
+	 */
+	protected static function get_selected_cron_events( $args, $assoc_args ) {
+		if ( empty( $args ) && ! Utils\get_flag_value( $assoc_args, 'due-now' ) && ! Utils\get_flag_value( $assoc_args, 'all' ) ) {
+			WP_CLI::error( 'Please specify one or more cron events, or use --due-now/--all.' );
+		}
```

Comment:
This also needs safeguards for flag combinations that don't make sense.
* If you provide hooks, the addition of `--due-now` or `--all` should throw an error.
* Providing both `--due-now` and `--all` at the same time should throw an error.

Reply:
Note: these should also have there seperate test cases.

Reply:
Yes, you are right. I will fix this soon.

Reply:
Hi @schlessera 

After seeing other tests break I realized it should be possible to select hooks and combine them with the `--due-now` parameter. Only hooks + `--all` and `--due-now --all` should be invalid.